dist/js/multiselect.js

```diff
@@ -1,755 +1,1812 @@
+ * @property {boolean} [submitAllLeft=true] - whether to submit all visible left options when a form is submitted
+ * @property {boolean} [submitAllRight=true] - whether to submit all visible right options when a form is submitted
+ * @property {SearchElements} [search] - the meta information used in the search elements
+ * @property {boolean} [ignoreDisabled=false] - whether to ignore disabled options when moving
+ * @property {string} [matchOptgroupBy='label'] - Which html attribute should be compared against when filtering
+ */
+
+/**
+ * Callback wrapper for the functions where Multiselect
+ * provides a way for the user to influence behaviour.
+ * @typedef {Object} CallbackFunctions
+ * @property {function} startUp - this function is called during initialisation, before the options are sorted
+ * @property {function} sort - used to define a different comparator from the default lexicographical comparison
+ * @property {function} beforeMoveToRight - called before the options are moved to the right side of the multiselect
+ * @property {function} afterMoveToRight - called after the options are moved to the right side of the multiselect
+ * @property {function} beforeMoveToLeft - called before the options are moved to the left side of the multiselect
+ * @property {function} afterMoveToLeft - called after the options are moved to the left side of the multiselect
+ * @property {function} move - called instead of the default move function
+ * @property {function} beforeMoveUp - called before options are moved up inside a right side select
+ * @property {function} afterMoveUp - called after options are moved up inside a right side select
+ * @property {function} beforeMoveDown - called before options are moved down inside a right side select
+ * @property {function} afterMoveDown - called after options are moved down inside a right side select
+ * @property {function} fireSearch - called to determine if the filter function for a side should be activated
+ */
+
+/**
+ * The settings object containing all possible options and information to create a Multiselect.
+ * @typedef {CallbackFunctions|ElementNames|MultiselectOptions} SettingsObject
+ */
+
+/**
+ * Describes options where the user provides inputs to use as select filters
+ * @typedef {Object} SearchElements
+ * @property {string|jQuery} [left] - either html for or a jQuery element pointing to the input to use for filtering the left palette
+ * @property {string|jQuery} [right] - either html for or a jQuery element pointing to the input to use for filtering the right palette
+ */
+
+/**
+ * The relation between a input used as a filter
+ * and the select elements that are filtered by it
+ * @typedef {Object} FilterRelation
+ * @property {jQuery} $filterInput - the input that you enter text in so that the select is filtered
+ * @property {jQuery} $filteredSelects - the select elements filtered by the input (currently always all right elements)
+ */
+
+/**
```

Comment:
Stack definition. I replaced the strings "left" and "right" with actual jQuery references so that its easier to undo/redo, and doable with multiple destinations.